Source/WebInspectorUI/UserInterface/Views/RecordingActionTreeElement.js

```diff
@@ -103,73 +101,63 @@ WI.RecordingActionTreeElement = class RecordingActionTreeElement extends WI.Gene
             case WI.Recording.Swizzle.WebGLSampler:
             case WI.Recording.Swizzle.WebGLSync:
             case WI.Recording.Swizzle.WebGLTransformFeedback:
-            case WI.Recording.Swizzle.WebGLVertexArrayObject:
-                parameterCopyText = WI.Recording.displayNameForSwizzleType(swizzleType);
-
-                parameterElement.textContent = parameterCopyText;
-                if (parameter) {
-                    let objectHandleElement = document.createElement("span");
-                    objectHandleElement.classList.add("parameter");
-                    objectHandleElement.classList.add("object-handle");
-                    objectHandleElement.textContent = `@${parameter}`;
-                    parameterElement.append(" ", objectHandleElement);
-                } else
+            case WI.Recording.Swizzle.WebGLVertexArrayObject: {
+                let parameterTypeText = WI.Recording.displayNameForSwizzleType(swizzleType);
+                if (!isNaN(parameter)) {
+                    parameterElement.classList.add("object-handle");
+                    parameterElement.textContent = `${parameterTypeText}${parameter + 1}`;
+                } else {
                     parameterElement.classList.add("swizzled");
+                    parameterElement.textContent = parameterTypeText;
+                }
                 break;
             }
+            }
 
             if (!parameterElement.textContent) {
                 parameterElement.classList.add("invalid");
                 parameterElement.textContent = swizzleType === WI.Recording.Swizzle.None ? parameter : WI.Recording.displayNameForSwizzleType(swizzleType);
             }
 
-            if (!parameterCopyText.length)
-                   parameterCopyText = parameterElement.textContent;
-
-            return {parameterElement, parameterCopyText};
+            return parameterElement;
         }
 
         let titleFragment = document.createDocumentFragment();
-        let copyText = recordingAction.name;
 
         let contextReplacer = recordingAction.contextReplacer;
         if (contextReplacer) {
-            copyText = contextReplacer + "." + copyText;
-
             let contextReplacerContainer = titleFragment.appendChild(document.createElement("span"));
             contextReplacerContainer.classList.add("context-replacer");
             contextReplacerContainer.textContent = contextReplacer;
+
+            titleFragment.appendChild(document.createTextNode("."));
```

Comment:
This code could generally be cleaner with `append()` instead of `appendChild()`:
```
let contextReplacerContainer = document.createElement("span");
...
titleFragment.append(contextReplacerContainer, ".", [...])
```

Not a big deal given most of the code is written in a different way... but that's generally how I would write this code.

Reply:
we've been doing it this way for a long time, but we could consider using `append` more

personally i like the `parent.appendChild(document.createElement("..."))` pattern as it makes it harder to forget to add the node to the DOM, but i can also see the benefit of having all the children added in one place